TestingHelpers/MockFile.cs

```diff
@@ -527,7 +527,23 @@ public override void SetAttributes(string path, FileAttributes fileAttributes)
         {
             mockFileDataAccessor.PathVerifier.IsLegalAbsoluteOrRelative(path, "path");
 
-            mockFileDataAccessor.GetFile(path).Attributes = fileAttributes;
+            var possibleFileData = mockFileDataAccessor.GetFile(path);
+            if (possibleFileData == null)
+            {
+                var directoryInfo = mockFileDataAccessor.DirectoryInfo.FromDirectoryName(path);
+                if (directoryInfo.Exists)
+                {
+                    directoryInfo.Attributes = fileAttributes;
+                }
+                else
+                {
+                    throw new FileNotFoundException($"Could not find file '{path}'.", path);
```

Comment:
I would take this opportunity to move this string into the Resources.resx file. Just make sure it's the same verbiage that System.IO uses, which it already looks like it is.

Reply:
As a followup, I'm about to deploy this resource verbiage as part of another PR.

Reply:
Ok, you should be able to use StringResources.Manager.GetString("COULD_NOT_FIND_FILE_EXCEPTION")

Reply:
I see what I'm supposed to be able to do, but the compiler keeps telling me:

```
'StringResources' is inaccessible due to its protection level
```

And I do see that you have this defined in the `AssemblyInfo.cs`:

```c#
using System.Runtime.CompilerServices;

#if DEBUG
[assembly: InternalsVisibleTo("System.IO.Abstractions.TestingHelpers.Tests")]
#else
    [assembly: InternalsVisibleTo("System.IO.Abstractions.TestingHelpers.Tests, PublicKey=002400000480000094000000060200000024000052534131000400000100010051bf2aa00ba30d507d4cebcab1751dfa13768a6f5235ce52da572260e33a11f52b87707f858fe4bbe32cd51830a8dd73245f688902707fa797c07205ff9b5212f93760d52f6d13022a286ff7daa13a0cd9eb958e888fcd7d9ed1f7cf76b19a5391835a7b633418a5f584d10925d76810f782f6b814cc34a2326b438abdc3b5bd")]
#endif
```

So, in theory, it should work. I even tried shortening the assembly name to be `TestingHelpers.Tests`, which I believe is actually what it should be, but that didn't seem to help.

It looks like I'm the first consumer of `StringResources` in the test project. That's why I'm the first to see this.

😭

Reply:
I'm not sure I follow. You're able to use it in the TestingHelpers library right? For MockInfo.cs? I wouldn't bother using it as an Assertion within the test. Just checking that it throws a FileNotFoundException is more than enough.